lib/src/components/progress/progress_bar.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import '../../../zeta_flutter.dart';
+import 'progress.dart';
+
+/// Enum for types of progress bar
+enum BarType {
+  /// Standard bar type.
+  standard,
+
+  /// Indeterminate bar type. positions randomly
+  indeterminate,
+
+  /// Buffering bar type. Has buffering at end.
+  buffering
+}
+
+/// BarWeight of progress bar
+enum BarWeight {
+  /// Thin weight.
+  thin,
+
+  ///Medium weight
+  medium
+}
+
+/// Linear progress bar.
+/// Uses progress percentage value to fill bar
+class ProgressBar extends Progress {
+  ///Constructor for [ProgressBar]
+
+  const ProgressBar(
+      {super.key,
+      required super.progress,
+      required this.border,
+      required this.type,
+      required this.weight,
+      this.label});
+
+  /// Constructs a standard progress bar
+  const ProgressBar.standard(
```

Comment:
These named constructors can remove un needed props, such as here type is not needed as it should always be standard, so we should change it to this:
```
  const ProgressBar.standard({
    super.key,
    required super.progress,
    this.border = ZetaWidgetBorder.rounded,
    this.weight = BarWeight.medium,
    this.label,
  }) : type = BarType.standard;
```